javascript-carousel/main.js

```diff
@@ -0,0 +1,184 @@
+var $leftArrow = document.querySelector('.fa-chevron-left');
+var $rightArrow = document.querySelector('.fa-chevron-right');
+var $allView = document.querySelectorAll('.view');
+var $allDotView = document.querySelectorAll('.dotView');
+var $dot1 = document.getElementById('1');
+var $dot2 = document.getElementById('2');
+var $dot3 = document.getElementById('3');
+var $dot4 = document.getElementById('4');
+var $dot5 = document.getElementById('5');
+var currentView = '1';
+var timerId = setInterval(handleRight, 3000);
+
+$leftArrow.addEventListener('click', handleLeft);
+$rightArrow.addEventListener('click', handleRight);
+$dot1.addEventListener('click', handleDot1);
+$dot2.addEventListener('click', handleDot2);
+$dot3.addEventListener('click', handleDot3);
+$dot4.addEventListener('click', handleDot4);
+$dot5.addEventListener('click', handleDot5);
```

Comment:
Instead of applying individual event handlers here, why not make use of event delegation? You can apply the listener to the parent and wait for an event originating from a "dot"